gossip/evm_state_reader.go

```diff
@@ -111,20 +114,28 @@ func (r *EvmStateReader) getBlock(h hash.Event, n idx.Block, readTxs bool) *evmc
 	}
 
 	// find block rules
-	epoch := block.Atropos.Epoch()
+	epoch := block.Epoch
 	es := r.store.GetHistoryEpochState(epoch)
 	var rules opera.Rules
 	if es != nil {
 		rules = es.Rules
 	}
-	var prev hash.Event
+
+	// There is no epoch state for epoch 0 comprising block 0.
+	// For this epoch, London and Sonic upgrades are enabled.
+	if epoch == 0 {
```

Comment:
was this change not also included in another PR ? I think it's a good change, just making sure its inclusion here is intended.

Reply:
Should not we include the epoch 0 state into the genesis file instead, like it is already done in this PR with the block 0?

Reply:
I tried getting this to work, but I could not.

I think the genesis generation process requires a general check-up. I created an issue for that and added a comment to the code that this requires a cleanup ([#72](https://github.com/Fantom-foundation/sonic-admin/issues/72)).